// gRPC는 Protocol Buffers (proto3) 문법을 사용함
// gRPC uses Protocol Buffers (proto3) syntax
syntax = "proto3";  // 프로토콜 버전 지정 (proto3 사용)

// 서비스 정의 (서버에서 제공하는 기능)
// Define the service (functions provided by the server)
service GameService {
    // GetPlayerState 메서드 정의 (요청을 보내면 응답을 반환함)
    // GetPlayerState method: Receives a request and returns a response
    rpc GetPlayerState (PlayerRequest) returns (PlayerResponse);
}

// 클라이언트가 서버에 보낼 데이터 형식 (요청 메시지)
// Define the request message structure (data sent by the client)
message PlayerRequest {
    string player_id = 1;  // 플레이어 ID (필수, 문자열)
}

// 서버가 클라이언트에게 보낼 데이터 형식 (응답 메시지)
// Define the response message structure (data returned by the server)
message PlayerResponse {
    string player_id = 1;  // 플레이어 ID
    optional float x = 2;           // 플레이어 위치 (x 좌표)
    optional float y = 3;           // 플레이어 위치 (y 좌표)
    int32 farm_level = 4;  // 플레이어의 농장 레벨
}
